tesseract/tesseract_kinematics/include/tesseract_kinematics/core/validate.h

```diff
@@ -58,40 +58,85 @@ inline bool checkKinematics(const tesseract_kinematics::ForwardKinematics::Const
 {
   // Check name
   if (fwd_kin->getName() != inv_kin->getName())
+  {
+    CONSOLE_BRIDGE_logError("checkKinematics: Manipulator names do not match!");
     return false;
+  }
 
   // Check if number of joints
   if (fwd_kin->numJoints() != inv_kin->numJoints())
+  {
+    CONSOLE_BRIDGE_logError("checkKinematics: Manipulator number of joints do not match!");
     return false;
+  }
 
   // Check link names
-  if (fwd_kin->getLinkNames().size() != inv_kin->getLinkNames().size() ||
-      !std::equal(fwd_kin->getLinkNames().begin(), fwd_kin->getLinkNames().end(), inv_kin->getLinkNames().begin()))
+  std::vector<std::string> l1 = fwd_kin->getLinkNames();
+  std::vector<std::string> l2 = inv_kin->getLinkNames();
+  std::sort(l1.begin(), l1.end());
+  std::sort(l2.begin(), l2.end());
+  if (l1.size() != l1.size())
+  {
+    CONSOLE_BRIDGE_logWarn("checkKinematics: Manipulator link names size do not match!");
```

Comment:
Should this be logError? It returns false.

Reply:
There could be a case where if you used a chain compared to a joint list the link list could be different. Even though they are different they may still produce the same adjacency map used for getting the active links. It should not return false, I will fix that.

Reply:
fixed.